hazelcast-client-new/src/main/java/com/hazelcast/client/proxy/ClientAtomicLongProxy.java

```diff
@@ -33,24 +33,26 @@
 import com.hazelcast.client.spi.ClientProxy;
 import com.hazelcast.core.IAtomicLong;
 import com.hazelcast.core.IFunction;
-import com.hazelcast.nio.serialization.Data;
 
 import static com.hazelcast.util.Preconditions.isNotNull;
 
 public class ClientAtomicLongProxy extends ClientProxy implements IAtomicLong {
 
-    private final String name;
-    private volatile Data key;
+    private int partitionId;
 
     public ClientAtomicLongProxy(String serviceName, String objectId) {
         super(serviceName, objectId);
-        this.name = objectId;
+    }
+
+    @Override
+    protected void onInitialize() {
+        partitionId = getContext().getPartitionService().getPartitionId(getName());
     }
 
     @Override
     public <R> R apply(IFunction<Long, R> function) {
         isNotNull(function, "function");
-        ClientMessage request = AtomicLongApplyCodec.encodeRequest(name, toData(function));
+        ClientMessage request = AtomicLongApplyCodec.encodeRequest(getName(), toData(function));
```

Comment:
Purely a flavor thing; I often make use of protected final field for this instead of doing a method call. Keeps the code a bit cleaner. But won't have any performance impact; jit will remove the getter anyway.